presto-main/src/main/java/com/facebook/presto/operator/scalar/MathFunctions.java

```diff
@@ -327,6 +331,69 @@ public static long truncate(@SqlType(StandardTypes.REAL) long num)
         return floatToRawIntBits((float) (Math.signum(numInFloat) * Math.floor(Math.abs(numInFloat))));
     }
 
+    @Description("truncate to double by dropping digits after decimal point")
+    @ScalarFunction
+    @SqlType(StandardTypes.DOUBLE)
+    public static double truncate(@SqlType(StandardTypes.DOUBLE) double num, @SqlType(StandardTypes.BIGINT) long decimals)
```

Comment:
We should probably change the type to be `INTEGER` since we're downcasting to int below.  Actually, why not make this `TINYINT` (max range is [-127, 128])?  If that's not sufficient precision, we could make it `SMALLINT` (range > 400) and safely downcast.

Reply:
are you referring to the type for decimals?it may not be sufficient to use `TINYINT` because truncate precision could have a long decimal precision. I  switch to`INTEGER` for now